### PR TITLE
Us35 49-51

### PR DESCRIPTION
**Pull request by BJSherman80**
- [x] done

User Story 35, Merchant Dashboard displays Orders

As a merchant employee
When I visit my merchant dashboard ("/merchant")
If any users have pending orders containing items I sell
Then I see a list of these orders.
Each order listed includes the following information:
-  [x] the ID of the order, which is a link to the order show page ("/merchant/orders/15")
- [x]  the date the order was made
- [x]  the total quantity of my items in the order
- [x] the total value of my items for that order

- [x]  done

User Story 49, Merchant sees an order show page

As a merchant employee
When I visit an order show page from my dashboard
I see the recipients name and address that was used to create this order
I only see the items in the order that are being purchased from my merchant
I do not see any items in the order being purchased from other merchants
For each item, I see the following information:
- the name of the item, which is a link to my item's show page
- an image of the item
- my price for the item
- the quantity the user wants to purchase

- [x] done

User Story 50, Merchant fulfills part of an order

As a merchant employee
When I visit an order show page from my dashboard
For each item of mine in the order
If the user's desired quantity is equal to or less than my current inventory quantity for that item
And I have not already "fulfilled" that item:
- Then I see a button or link to "fulfill" that item
- When I click on that link or button I am returned to the order show page
- I see the item is now fulfilled
- I also see a flash message indicating that I have fulfilled that item
- the item's inventory quantity is permanently reduced by the user's desired quantity

If I have already fulfilled this item, I see text indicating such.

- [x] done

User Story 51, Merchant cannot fulfill an order due to lack of inventory

As a merchant employee
When I visit an order show page from my dashboard
For each item of mine in the order
If the user's desired quantity is greater than my current inventory quantity for that item
Then I do not see a "fulfill" button or link
Instead I see a notice next to the item indicating I cannot fulfill this item

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -59,6 +59,8 @@
   namespace :merchant do
     get '/', to: 'dashboard#show'
     get '/items', to: 'items#index'
+    resources :item_orders, only: [:update]
+    resources :orders, only: [:show]
```
Comment (helloeduardo) [inline config/routes.rb:63]: Great job adding more namespaced resources here

**Review by helloeduardo — APPROVED** [review]
Looks good!

File: app/models/merchant.rb [hunk 2]
```diff
@@ -26,7 +27,7 @@ def distinct_cities
     item_orders.distinct.joins(:order).pluck(:city)
   end
 
-  def full_address
-    "#{address}, #{city}, #{state} #{zip}"
+  def pending_orders
+    orders.where(status: "pending")
```
Comment (helloeduardo) [inline app/models/merchant.rb:31]: Nice use of AR here

File: spec/models/merchant_spec.rb [hunk 3]
```diff
@@ -11,7 +11,9 @@
 
   describe "relationships" do
     it { should have_many :items }
+    it { should have_many(:item_orders).through(:items) }
```
Comment (the-color-bliu) [inline spec/models/merchant_spec.rb:14]: Looks like this works, isn't the syntax here normally supposed to be ``it { should have_many(:item_orders) }`` and ``it { should have_many(:items).through(:item_orders)``?

**Review by the-color-bliu — APPROVED** [review]
Everything looks good here, great job!

**Pull request merged**